cvat-core/src/index.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import PluginRegistry from './plugins';
+import serverProxy from './server-proxy';
+import lambdaManager from './lambda-manager';
+import { AnnotationFormats } from './annotation-formats';
+import loggerStorage from './logger-storage';
+import * as enums from './enums';
+import config from './config';
+import { mask2Rle, rle2Mask } from './object-utils';
+import User from './user';
+import Project from './project';
+import { Job, Task } from './session';
+import { EventLogger } from './log';
+import { Attribute, Label } from './labels';
+import Statistics from './statistics';
+import ObjectState from './object-state';
+import MLModel from './ml-model';
+import Issue from './issue';
+import Comment from './comment';
+import { FrameData } from './frames';
+import CloudStorage from './cloud-storage';
+import Organization from './organization';
+import Webhook from './webhook';
+import AnnotationGuide from './guide';
+import {
+    ArgumentError, DataError, Exception, ScriptingError, ServerError,
+} from './exceptions';
+
+export default interface CVATCore {
+    plugins: {
+        list: typeof PluginRegistry.list;
+        register: typeof PluginRegistry.register;
+    };
+    lambda: {
+        list: typeof lambdaManager.list;
+        run: typeof lambdaManager.run;
+        call: typeof lambdaManager.call;
+        cancel: typeof lambdaManager.cancel;
+        listen: typeof lambdaManager.listen;
+        requests: typeof lambdaManager.requests;
+    };
+    server: {
+        about: typeof serverProxy.server.about;
+        share: (dir: string) => Promise<{
+            mimeType: string;
+            name: string;
+            type: enums.ShareFileType;
+        }[]>;
+        formats: () => Promise<AnnotationFormats>;
+        userAgreements: typeof serverProxy.server.userAgreements,
+        register: any; // TODO: add types later
```

Comment:
Will it be added in this patch or wait for next one?

Reply:
I was planning to add these types gradually. It is too big task for one PR